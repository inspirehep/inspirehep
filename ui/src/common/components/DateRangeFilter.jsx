import React, {
  useCallback,
  useMemo,
} from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import { DatePicker, Row } from 'antd';

import {
  RANGE_AGGREGATION_SELECTION_SEPARATOR,
  DATE_RANGE_FORMAT,
} from '../constants';

function DateRangeFilter({ onChange, range = '' }) {
  const [startDate = '', endDate = ''] = useMemo(
    () => range.split(RANGE_AGGREGATION_SELECTION_SEPARATOR),
    [range]
  );

  const startDateMoment = useMemo(
    () => (startDate ? moment.utc(startDate, DATE_RANGE_FORMAT) : null),
    [startDate]
  );

  const endDateMoment = useMemo(
    () => (endDate ? moment.utc(endDate, DATE_RANGE_FORMAT) : null),
    [endDate]
  );

  const isLaterThanEndDate = useCallback(
    dateAsMoment => {
      if (!dateAsMoment || !endDateMoment) {
        return false;
      }
      return dateAsMoment.isSameOrAfter(endDateMoment);
    },
    [endDateMoment]
  );

  const isEarlierThanStartDate = useCallback(
    dateAsMoment => {
      if (!dateAsMoment || !startDateMoment) {
        return false;
      }
      return dateAsMoment.isSameOrBefore(startDateMoment);
    },
    [startDateMoment]
  );

  const onStartDateChange = useCallback(
    (_, dateString) => {
      const willHaveAnySelectedDate = dateString || endDate;
      onChange(
        willHaveAnySelectedDate
          ? `${dateString}${RANGE_AGGREGATION_SELECTION_SEPARATOR}${endDate}`
          : undefined
      );
    },
    [endDate, onChange]
  );

  const onEndDateChange = useCallback(
    (_, dateString) => {
      const willHaveAnySelectedDate = dateString || startDate;
      onChange(
        willHaveAnySelectedDate
          ? `${startDate}${RANGE_AGGREGATION_SELECTION_SEPARATOR}${dateString}`
          : undefined
      );
    },
    [startDate, onChange]
  );

  return (
    <div>
      <Row className="mb1">From:</Row>
      <Row className="mb2">
        <DatePicker
          data-test-id="start-date-picker"
          format={DATE_RANGE_FORMAT}
          className="w-100"
          onChange={onStartDateChange}
          value={startDateMoment}
          disabledDate={isLaterThanEndDate}
        />
      </Row>
      <Row className="mb1">To:</Row>
      <Row>
        <DatePicker
          data-test-id="end-date-picker"
          format={DATE_RANGE_FORMAT}
          onChange={onEndDateChange}
          className="w-100"
          value={endDateMoment}
          disabledDate={isEarlierThanStartDate}
        />
      </Row>
    </div>
  );
}

DateRangeFilter.propTypes = {
  onChange: PropTypes.func.isRequired,
  range: PropTypes.string,
};

export default DateRangeFilter;
