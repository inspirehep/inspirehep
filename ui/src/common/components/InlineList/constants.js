export const DEFAULT_SEPARATE_ITEMS_CLASS = 'separate-items-with-comma';
export const SEPARATE_ITEMS_CLASSNAMES = [
  'separate-items-with-semicolon',
  'separate-items-with-comma',
  'separate-items-with-and',
  'separate-items-with-middledot',
];
