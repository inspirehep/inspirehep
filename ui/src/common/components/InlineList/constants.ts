export const SEPARATOR_SEMICOLON = '; ';
export const SEPARATOR_COMMA = ', ';
export const SEPARATOR_AND = ' and ';
export const SEPARATOR_MIDDLEDOT = ' \u2022 ';

export const SEPARATOR_TYPES = [
  SEPARATOR_SEMICOLON,
  SEPARATOR_COMMA,
  SEPARATOR_AND,
  SEPARATOR_MIDDLEDOT,
];

export const DEFAULT_SEPARATOR_TYPE = SEPARATOR_COMMA;
