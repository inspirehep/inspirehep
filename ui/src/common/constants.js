export const PUBLISHED_QUERY = { citeable: true, refereed: true };
export const CITEABLE_QUERY = { citeable: true, refereed: undefined };
export const PUBLISHED_BAR_TYPE = 'published';
export const CITEABLE_BAR_TYPE = 'citeable';

export const POST_DOC_RANK_VALUE = 'POSTDOC';
export const RANK_VALUE_TO_DISPLAY = {
  SENIOR: 'Senior (permanent)',
  JUNIOR: 'Junior (leads to Senior)',
  STAFF: 'Staff (non-research)',
  VISITOR: 'Visitor',
  [POST_DOC_RANK_VALUE]: 'PostDoc',
  PHD: 'PhD',
  MASTER: 'Master',
  UNDERGRADUATE: 'Undergrad',
  OTHER: 'Other',
};

export const AUTHORS_PID_TYPE = 'authors';
export const LITERATURE_PID_TYPE = 'literature';
export const JOBS_PID_TYPE = 'jobs';
