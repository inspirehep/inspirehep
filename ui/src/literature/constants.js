export const CITE_FORMAT_OPTIONS = [
  { value: 'x-bibtex', display: 'BibTeX' },
  { value: 'vnd+inspire.latex.eu+x-latex', display: 'LaTeX (EU)' },
  { value: 'vnd+inspire.latex.us+x-latex', display: 'LaTeX (US)' },
];
export const CITE_FORMAT_VALUES = CITE_FORMAT_OPTIONS.map(
  option => option.value
);

export const MAX_CITEABLE_RECORDS = 1000;
