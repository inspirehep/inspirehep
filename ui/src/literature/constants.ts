const BIBTEX = 'application/x-bibtex';
const LATEX_EU = 'application/vnd+inspire.latex.eu+x-latex';
const LATEX_US = 'application/vnd+inspire.latex.us+x-latex';
export const CV = 'text/vnd+inspire.html+html';

export const CITE_FORMAT_OPTIONS = [
  { value: BIBTEX, display: 'BibTeX' },
  { value: LATEX_EU, display: 'LaTeX (EU)' },
  { value: LATEX_US, display: 'LaTeX (US)' },
  { value: CV, display: 'CV (html)' },
];

export const CITE_FILE_FORMAT = {
  [BIBTEX]: { mimetype: 'application/x-bibtex', extension: 'bib' },
  [LATEX_EU]: { mimetype: 'application/x-latex', extension: 'tex' },
  [LATEX_US]: { mimetype: 'application/x-latex', extension: 'tex' },
  [CV]: { mimetype: 'text/html', extension: 'html' },
};

export const CITE_FORMAT_VALUES = CITE_FORMAT_OPTIONS.map(
  (option) => option.value
);

export const MAX_CITEABLE_RECORDS = 1000;

export const MAX_BULK_ASSIGN = 250;
