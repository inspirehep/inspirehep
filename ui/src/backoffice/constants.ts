import { List, Map } from 'immutable';

import { AUTHORS_PID_TYPE, LITERATURE_PID_TYPE } from '../common/constants';

export enum WorkflowTypes {
  AUTHOR_CREATE = 'AUTHOR_CREATE',
  AUTHOR_UPDATE = 'AUTHOR_UPDATE',
  HEP_CREATE = 'HEP_CREATE',
}

export const WORKFLOW_TYPES = {
  [WorkflowTypes.AUTHOR_CREATE]: AUTHORS_PID_TYPE,
  [WorkflowTypes.AUTHOR_UPDATE]: AUTHORS_PID_TYPE,
  [WorkflowTypes.HEP_CREATE]: LITERATURE_PID_TYPE,
};

export interface WorkflowStatus {
  key: string;
  doc_count: number;
}

export interface WorkflowType {
  key: string;
  doc_count: number;
}

export interface Collection {
  key: string;
  value: string | undefined;
}

export interface WorkflowCardProps {
  type: Map<string, any>;
  statuses: List<Map<string, any>>;
}

export type WorkflowTypeKey = 'AUTHOR_CREATE' | 'AUTHOR_UPDATE' | 'HEP_CREATE';
