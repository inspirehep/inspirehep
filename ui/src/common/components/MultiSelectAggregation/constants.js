import { RANK_VALUE_TO_DISPLAY } from '../../constants';

// moved separately from the component file to make it easy to mock in tests
export const SELECT_VALUE_TO_DISPLAY_MAPS_FOREACH_AGG = {
  Rank: RANK_VALUE_TO_DISPLAY,
};
