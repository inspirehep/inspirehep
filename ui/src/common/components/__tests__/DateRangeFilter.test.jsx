import React from 'react';
import { shallow } from 'enzyme';
import moment from 'moment';
import { advanceTo, clear } from 'jest-date-mock';
import DateRangeFilter from '../DateRangeFilter';
import { DATE_RANGE_FORMAT } from '../../constants';

describe('DateRangeFilter', () => {
  afterEach(() => {
    clear();
  });

  it('renders DateRangeFilter with all props set', () => {
    const wrapper = shallow(
      <DateRangeFilter
        onChange={jest.fn()}
        range="2019-11-21--2019-11-22"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders DateRangeFilter without range', () => {
    const wrapper = shallow(
      <DateRangeFilter onChange={jest.fn()} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('calls onChange when start date selected and no range', () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <DateRangeFilter onChange={onChange} />
    );
    const currentDate = advanceTo(new Date('2019-05-28T13:31:00+00:00'));
    const onStartDateChanged = wrapper
      .find('[data-test-id="start-date-picker"]')
      .prop('onChange');
    onStartDateChanged(
      moment(currentDate),
      moment(currentDate).format(DATE_RANGE_FORMAT)
    );
    expect(onChange).toHaveBeenCalledWith('2019-05-28--');
  });

  it('calls onChange when start date selected and there are range', () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <DateRangeFilter
        onChange={onChange}
        range="2019-11-21--2019-11-22"
      />
    );
    const currentDate = advanceTo(new Date('2019-05-28T13:31:00+00:00'));
    const onStartDateChanged = wrapper
      .find('[data-test-id="start-date-picker"]')
      .prop('onChange');
    onStartDateChanged(
      moment(currentDate),
      moment(currentDate).format(DATE_RANGE_FORMAT)
    );
    expect(onChange).toHaveBeenCalledWith('2019-05-28--2019-11-22');
  });

  it('calls onChange when end date selected and there are range', () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <DateRangeFilter
        onChange={onChange}
        range="2019-04-21--2019-11-22"
      />
    );
    const currentDate = advanceTo(new Date('2019-05-28T13:31:00+00:00'));
    const onStartDateChanged = wrapper
      .find('[data-test-id="end-date-picker"]')
      .prop('onChange');
    onStartDateChanged(
      moment(currentDate),
      moment(currentDate).format(DATE_RANGE_FORMAT)
    );
    expect(onChange).toHaveBeenCalledWith('2019-04-21--2019-05-28');
  });

  it('calls onChange when end date selected and no range', () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <DateRangeFilter onChange={onChange} />
    );
    const currentDate = advanceTo(new Date('2019-05-28T13:31:00+00:00'));
    const onStartDateChanged = wrapper
      .find('[data-test-id="end-date-picker"]')
      .prop('onChange');
    onStartDateChanged(
      moment(currentDate),
      moment(currentDate).format(DATE_RANGE_FORMAT)
    );
    expect(onChange).toHaveBeenCalledWith('--2019-05-28');
  });

  it('calls onChange with cleared start date when start date is removed and there are range', () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <DateRangeFilter
        onChange={onChange}
        range="2019-04-21--2019-11-22"
      />
    );
    const onStartDateChanged = wrapper
      .find('[data-test-id="start-date-picker"]')
      .prop('onChange');
    onStartDateChanged(null, '');
    expect(onChange).toHaveBeenCalledWith('--2019-11-22');
  });

  it('calls onChange with cleared start date when start date is removed and no range', () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <DateRangeFilter onChange={onChange} />
    );
    const onStartDateChanged = wrapper
      .find('[data-test-id="start-date-picker"]')
      .prop('onChange');
    onStartDateChanged(null, '');
    expect(onChange).toHaveBeenCalledWith(undefined);
  });

  it('calls onChange with cleared end date when end date is removed and there are range', () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <DateRangeFilter
        onChange={onChange}
        range="2019-04-21--2019-11-22"
      />
    );
    const onStartDateChanged = wrapper
      .find('[data-test-id="end-date-picker"]')
      .prop('onChange');
    onStartDateChanged(null, '');
    expect(onChange).toHaveBeenCalledWith('2019-04-21--');
  });

  it('calls onChange with cleared end date when end date is removed and no range', () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <DateRangeFilter onChange={onChange} />
    );
    const onStartDateChanged = wrapper
      .find('[data-test-id="end-date-picker"]')
      .prop('onChange');
    onStartDateChanged(null, '');
    expect(onChange).toHaveBeenCalledWith(undefined);
  });

  it('sets start date disabled when end date before given date', () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <DateRangeFilter
        onChange={onChange}
        range="--2019-11-22"
      />
    );
    const disabledDate = wrapper
      .find('[data-test-id="start-date-picker"]')
      .prop('disabledDate');
    expect(disabledDate(moment(new Date('2019-11-28T13:31:00+00:00')))).toBe(
      true
    );
  });

  it('does not set start date disabled when end date is later than given date', () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <DateRangeFilter
        onChange={onChange}
        range="--2019-11-22"
      />
    );
    const disabledDate = wrapper
      .find('[data-test-id="start-date-picker"]')
      .prop('disabledDate');
    expect(disabledDate(moment(new Date('2019-11-21T13:31:00+00:00')))).toBe(
      false
    );
  });

  it('sets end date disabled when start date after given date', () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <DateRangeFilter
        onChange={onChange}
        range="2019-11-22--"
      />
    );
    const disabledDate = wrapper
      .find('[data-test-id="end-date-picker"]')
      .prop('disabledDate');
    expect(disabledDate(moment(new Date('2019-11-21T13:31:00+00:00')))).toBe(
      true
    );
  });

  it('does not set end date disabled when start date is before than given date', () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <DateRangeFilter
        onChange={onChange}
        range="2019-11-22--"
      />
    );
    const disabledDate = wrapper
      .find('[data-test-id="end-date-picker"]')
      .prop('disabledDate');
    expect(disabledDate(moment(new Date('2019-11-29T13:31:00+00:00')))).toBe(
      false
    );
  });

  it('does not set start date disabled when there is no end date', () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <DateRangeFilter onChange={onChange} />
    );
    const disabledDate = wrapper
      .find('[data-test-id="start-date-picker"]')
      .prop('disabledDate');
    expect(disabledDate(moment(new Date('2019-11-21T13:31:00+00:00')))).toBe(
      false
    );
  });

  it('does not set end date disabled when there is no start date', () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <DateRangeFilter onChange={onChange} />
    );
    const disabledDate = wrapper
      .find('[data-test-id="end-date-picker"]')
      .prop('disabledDate');
    expect(disabledDate(moment(new Date('2019-11-21T13:31:00+00:00')))).toBe(
      false
    );
  });
});
