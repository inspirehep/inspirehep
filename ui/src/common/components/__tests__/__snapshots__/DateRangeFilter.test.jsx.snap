// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateRangeFilter renders DateRangeFilter with all props set 1`] = `
<div>
  <Row
    className="mb1"
  >
    From:
  </Row>
  <Row
    className="mb2"
  >
    <ForwardRef
      className="w-100"
      data-test-id="start-date-picker"
      disabledDate={[Function]}
      format="YYYY-MM-DD"
      onChange={[Function]}
      value={"2019-11-21T00:00:00.000Z"}
    />
  </Row>
  <Row
    className="mb1"
  >
    To:
  </Row>
  <Row>
    <ForwardRef
      className="w-100"
      data-test-id="end-date-picker"
      disabledDate={[Function]}
      format="YYYY-MM-DD"
      onChange={[Function]}
      value={"2019-11-22T00:00:00.000Z"}
    />
  </Row>
</div>
`;

exports[`DateRangeFilter renders DateRangeFilter without range 1`] = `
<div>
  <Row
    className="mb1"
  >
    From:
  </Row>
  <Row
    className="mb2"
  >
    <ForwardRef
      className="w-100"
      data-test-id="start-date-picker"
      disabledDate={[Function]}
      format="YYYY-MM-DD"
      onChange={[Function]}
      value={null}
    />
  </Row>
  <Row
    className="mb1"
  >
    To:
  </Row>
  <Row>
    <ForwardRef
      className="w-100"
      data-test-id="end-date-picker"
      disabledDate={[Function]}
      format="YYYY-MM-DD"
      onChange={[Function]}
      value={null}
    />
  </Row>
</div>
`;
